pilot/spec/tester_manager_spec.rb

```diff
@@ -104,5 +63,191 @@
+
+        expect(Spaceship::Tunes::Tester::External).to_not receive(:create!)
+        expect(Spaceship::TestFlight::Group).to receive(:add_tester_to_groups!).and_return([custom_tester_group])
+        expect(FastlaneCore::UI).to receive(:success).with('Successfully added tester to group(s): Test Group in app: My Fake App')
+
+        tester_manager.add_tester(default_add_tester_options_with_group)
+      end
+    end
+
+    describe "when asked to add an existing external tester to a specific existing custom group" do
+      it "adds the tester to the custom group" do
+        allow(current_user).to receive(:roles).and_return(["appmanager"])
+        allow(tester_manager).to receive(:find_app_tester).and_return(fake_tester)
+
+        expect(Spaceship::TestFlight::Tester).to_not receive(:create_app_level_tester)
+        expect(Spaceship::TestFlight::Group).to receive(:add_tester_to_groups!).and_return([custom_tester_group])
+        expect(FastlaneCore::UI).to receive(:success).with('Successfully added tester to group(s): Test Group in app: My Fake App')
+
+        tester_manager.add_tester(default_add_tester_options_with_group)
+      end
+    end
+
+    describe "when asked to add an existing internal tester to a specific existing custom group" do
+      it "adds the tester to the custom group" do
+        allow(current_user).to receive(:roles).and_return(["appmanager"])
+        allow(fake_tester).to receive(:kind_of?).with(Spaceship::Tunes::Tester::Internal).and_return(true)
+
+        expect(Spaceship::Tunes::Tester::Internal).to receive(:find_by_app).and_return(fake_tester)
+        expect(Spaceship::TestFlight::Tester).to_not receive(:create_app_level_tester)
+        expect(Spaceship::TestFlight::Group).to receive(:add_tester_to_groups!).and_return([custom_tester_group])
+        expect(FastlaneCore::UI).to receive(:success).with('Found existing tester fabric-devtools@gmail.com+fake@gmail.com')
+        expect(FastlaneCore::UI).to receive(:success).with('Successfully added tester to app My Fake App')
+
+        tester_manager.add_tester(default_add_tester_options_with_group)
+      end
+    end
+
+    describe "when app manager asks to add an existing external tester to a specific existing custom group" do
+      it "adds the tester without calling create" do
+        allow(current_user).to receive(:roles).and_return(["appmanager"])
+        allow(tester_manager).to receive(:find_app_tester).and_return(fake_tester)
+
+        expect(Spaceship::TestFlight::Tester).to_not receive(:create_app_level_tester)
+        expect(Spaceship::TestFlight::Group).to receive(:add_tester_to_groups!).and_return([custom_tester_group])
+        expect(FastlaneCore::UI).to receive(:success).with('Successfully added tester to group(s): Test Group in app: My Fake App')
+
+        tester_manager.add_tester(default_add_tester_options_with_group)
+      end
+    end
+
+    # Data and mocks
+    let(:tester_manager) { Pilot::TesterManager.new }
```

Comment:
Ohhh, will fix. I like having that stuff at the bottom so I can see the test logic. That's usually what I care about the most.